aws/resource_aws_client_vpn_endpoint.go

```diff
@@ -0,0 +1,279 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsClientVpnEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsClientVpnEndpointCreate,
+		Read:   resourceAwsClientVpnEndpointRead,
+		Delete: resourceAwsClientVpnEndpointDelete,
+		Update: resourceAwsClientVpnEndpointUpdate,
+		Importer: &schema.ResourceImporter{
```

Comment:
I'd add a Timeout block for the creation/deletion `WaitForState` calls (see below):

```go
		Timeouts: &schema.ResourceTimeout{
			Create: schema.DefaultTimeout(5 * time.Minute),
			Delete: schema.DefaultTimeout(5 * time.Minute),
		},
```